tests/test_binary.nim

```diff
@@ -7,6 +7,260 @@ const
     else: ""
   binaryName = "configlet" & binaryExt
 
+proc cloneExercismRepo(repoName, dest: string; isShallow = false): tuple[
+    output: string; exitCode: int] =
+  # Clones the Exercism repo named `repoName` to the location `dest`.
+  let opts = if isShallow: "--depth 1" else: ""
+  let url = &"https://github.com/exercism/{repoName}/"
+  let cmd = &"git clone {opts} {url} {dest}"
+  result = execCmdEx(cmd)
+
+template execAndCheck(expectedExitCode: int; body: untyped) {.dirty.} =
+  ## Runs `body`, and prints the output if the exit code is non-zero.
+  ## `body` must have the same return type as `execCmdEx`.
+  let (outp, exitCode) = body
+  if exitCode != expectedExitCode:
+    echo outp
+    fail()
+
+func conciseDiff(s: string): string =
+  ## Returns the lines of `s` that begin with a `+` or `-` character.
+  result = newStringOfCap(s.len)
+  for line in s.splitLines():
+    if line.len > 0:
+      if line[0] in {'+', '-'}:
```

Comment:
Minor style suggestion: I prefer to reduce nesting when possible, or is that against the Nim style guide?

```suggestion
    if line.len > 0 and line[0] in {'+', '-'}:
```

Reply:
There isn't an official style rule one way or the other. I'm fine with either way, but lately I've been leaning towards extra nesting.

There is at least [one (opinionated) Nim style guide](https://github.com/disruptek/disruptek/blob/decfd8711a1f49e5da279e65abcb9f3245721099/style.md#super-dominating-conditionals) that argues that nesting conditionals is more readable. I'll let you decide :)

A little related is a discussion between
```Nim
if not cond:
  return

foo1
foo2
foo3
```

vs

```Nim
if cond:
  foo1
  foo2
  foo3
```

where I can see the argument to prefer the latter - see Araq's post in https://forum.nim-lang.org/t/7130

Reply:
🤷 I'm fine with keeping the indentation here. We'll just evaluate per case what the best nesting is.